### DISCUSSION TITLE
Feature Request: Ability to approve workflow runs from the mobile app

### DISCUSSION THREAD

**Discussion by deadlydog** [community/community#110751] [Mobile]
### Select Topic Area

Product Feedback

### Body

GitHub has `Environments` that allow you to add `Required reviewers`, which effectively can block a workflow run until it is manually reviewed/approved. Here's [the docs](https://docs.github.com/en/actions/managing-workflow-runs/reviewing-deployments). This is a critical feature for many enterprise organizations, allowing them to block deployments to ensure it does not go out until the appropriate time.

Currently, the reviewers will get a notification that they need to approve the workflow run, but they are not able to approve it from the mobile app. They can view the workflow run, but there is no button to approve it. Instead, you currently need to log into the GitHub website from your phone (which often tries to redirect you back into the app) and approve the workflow run from there. The workflow run website page is not optimized for mobile though, so it's a bit of a pain to get to the button.

It would be great if the mobile app allowed you to approve workflow runs without needing to go to the website.

**Comment by alcere** [comment]
@deadlydog Thanks for your feedback. Approving deployments should be possible using the mobile app. Can you share with us the platform and app version you are using? I'd like to investigate this issue deeper. Thanks!

**Comment by deadlydog** [comment]
Hi @alcere , I'm using an iPhone 12 Pro iOS 17.3.1, with the latest GitHub App version 1.151.1.

Here's a screenshot of [the workflow run](https://github.com/deadlydog/PowerShell.tiPS/actions/runs/8143085610) shown in the app, either when clicking the "Review pending deployments" link in the approval email, or when manually navigating to the GitHub Actions workflow app manually in the app:

![GitHub run approval missing (Small)](https://github.com/community/community/assets/1187140/b809978d-96ca-4137-9a51-657bc75d2f52)

The only button that's available to click is "Cancel workflow"; there is no Review or Approve button.

Here's what the same workflow run looks like when I open it up in the browser (Edge in this case):

![GitHub run from website (Small)](https://github.com/community/community/assets/1187140/048bea22-aff1-4520-bdaa-4f16d6375b21)

And this is what the approval on the website looks like when I click the "Review deployments" button:

![GitHub approval from website (Small)](https://github.com/community/community/assets/1187140/cd3786c1-379e-4ac9-bdc3-662ce47d3572)

Since I'm signed into the same account on both and everything works fine from the website, it should not be a permission issue.

**Reply by alcere** [reply]
Oh, I see! We only support opening the Deployment review dialog using notifications (Push notifications, or within the Notifications section) in the GitHub mobile app. But I totally understand your problem here; it should be possible too using the link or from the GitHub Action itself.  I will open an internal issue to follow up on this and consider adding the option to approve deployments from those sources. Thank you for your feedback!

**Comment by deadlydog** [comment]
Thanks alcere. It's good to know that it's possible to approve from the notifications. I don't see any review/approval notifications for my deployments in the app, but I have disabled all of the push notifications within the app. I'm guessing if we disable the push notifications then that prevents them from showing up in the notifications section at all? I guess in the meantime, I'll enable the `Deployment review` push notification and see if approval notifications start showing up. Thanks!

**Reply by alcere** [reply]
Hey @deadlydog, I am sorry for the late response. Your push notification configuration shouldn't affect the type of notifications you receive. Were you able to receive review/approval notifications ?

**Reply by deadlydog** [reply]
Hey @alcere , I just updated to the latest version of the app (v1.183.0) on my iPhone and tested again. The behaviour is still the same as my original report. If I launch the app from the push notification then I see the approve button, but if I launch it from the link in the email notification or just open the GitHub app and navigate the Action manually then I only have a "Cancel workflow" button. If I navigate to the Notifications menu in the app, I do not have a notification for the Approval. If I have push notifications disabled (specifically the "Deployment Review" push notification type in the app), or I miss the push notification, the only way to approve it is to navigate to the Action in a web browser.

So in short, I don't see any change in behaviour since my original report of this issue.

**Reply by alcere** [reply]
Thanks for the details @deadlydog. Everything you have described seems to expected, except:

> If I navigate to the Notifications menu in the app, I do not have a notification for the Approval.

Just to double check:

You should be able to see there a notification similar to this:

![Screenshot 2024-10-28 at 12 00 10](https://github.com/user-attachments/assets/c8475e07-991f-4241-bbac-9137c73ddbb9)

Can you confirm that you don't see it?

Thanks!

**Reply by deadlydog** [reply]
Thanks for following up @alcere . No, I do not see that notification. [In this GitHub Action workflow run](https://github.com/deadlydog/PowerShell.tiPS/actions/runs/11502876289) you can see the workflow ran on October 24th at 9:49am CST and I approved it at 10:09am CST.

![image](https://github.com/user-attachments/assets/af9fbed0-1531-49b8-8163-68df1d74f632)

Looking in the iPhone GitHub app notifications screen, you can see I received a notification about the pull request being created on Oct 24th and merged in, but nothing about a deployment review approval once the PR was merged in:

![GitHub app notifications](https://github.com/user-attachments/assets/a052c3df-dd31-4b84-9f05-7162ed9ba120)

These are my GitHub app notification settings currently:

![GitHub app notification settings](https://github.com/user-attachments/assets/387312dd-7479-4210-8513-fc1b69e94c33)

The only reason I have the "Deployment Review" push notification enabled is to work around this problem.

Hopefully this helps.

**Reply by alcere** [reply]
Thanks for the follow-up, and I apologize for the late response. Unfortunately, I've been unavailable for the last couple of months. Do you see these types of notifications (for deployment reviews) using  https://github.com/? To do so, can you double check in https://github.com/settings/notifications under the `System` section if you got Notifications for workflow runs enabled for that particular repository? This shouldn't be necessary to see push notification for deployment review, but I'd like to double check that you are seeing app notifications for deployment reviews and pinpoint the issue.

**Reply by deadlydog** [reply]
No, I do not see the notifications in github.com either. I had [a deployment 2 weeks ago](https://github.com/deadlydog/PowerShell.tiPS/actions/runs/12511913583) and it does not show up in the mobile app notifications or the website notifications:

![image](https://github.com/user-attachments/assets/2e4c56dd-dd54-4f05-9e33-797067184507)

These are my current system notification settings:

![image](https://github.com/user-attachments/assets/7278135b-47f1-49c4-9634-b9acda5e75d5)

I can enable "On GitHub" notifications in addition to the Email notifications for Actions if you need; just let me know.

**Reply by alcere** [reply]
Yes please, give it a try enabling "On GitHub" and let me know the result please.

**Reply by deadlydog** [reply]
After enabling "On GitHub" and unchecking "Only notify for failed workflows" it now shows up in the GitHub.com notifications, but still does not show up in the mobile app notifications screen.

![image](https://github.com/user-attachments/assets/197a3b17-3bda-475a-a670-fa0f7c894b86)

**Reply by alcere** [reply]
Thanks @deadlydog for your help! Certainly this looks like the problem to me. Please, let me do some investigations. I will reach you back as soon as possible!

**Comment by sam-utila** [comment]
Any news regarding this?

This is a crucial workflow - especially for reviewing access for teammates on production environments, which we would like to do using the mobile app.

**Comment by mccrearyp** [comment]
I also need this feature

**Comment by YungBS** [comment]
This is a great idea, to develop the mobile app 💲✅🚀📍🧲

**Comment by Thepitufo13** [comment]
Yes i agree

**Comment by charlescrain** [comment]
Agree this would be a very helpful feature

**Comment by YungBS** [comment]
https://discord.gg/gQGmxJHf?event=1336993171648942160

**Reply by ssbarnea** [reply]
Still broken, iphone. Unable to merge due to oath app lacking workflow permission but there is no option in the app to change the authentication token!

**Comment by ioannisj** [comment]
+1

**Comment by matthewdowney** [comment]
Please prioritize this feature! :)

**Comment by danvy** [comment]
840 days and counting 🤔

**Comment by p-poppe** [comment]
hi any updates on this topic?

**Comment by payaaam** [comment]
Don't you guys have enough engineers and computer to one shot this at this point?

Also, can you please allow us to force approve PRs if we have correct permissions? Can you guys post a PR this weekend for these fixes?

It's truly incredible that these basic app features aren't done considering Microsoft and GitHub's partnership with AI labs
